### PR TITLE
Verifying that tools do not assume sh is in /bin/sh

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/durable-task-plugin/pull/95. @Vlatombe

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -91,10 +91,10 @@
       <version>6.7</version>
     </dependency>
 
-    <dependency> <!-- OnceRetentionStrategy -->
+    <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>durable-task</artifactId>
-      <version>1.28</version>
+      <version>1.30-SNAPSHOT</version> <!-- TODO https://github.com/jenkinsci/durable-task-plugin/pull/95 -->
```
Comment (jglick) [inline pom.xml]: pending incremental deployment

File: src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/bourneShellElsewhereInPath.groovy [hunk 2]
```diff
@@ -0,0 +1,7 @@
+podTemplate(label: 'bourneShellElsewhereInPath', containers: [containerTemplate(name: 'kaniko', image: 'gcr.io/kaniko-project/executor:debug', command: 'sleep', args: '99d')]) {
+    node('bourneShellElsewhereInPath') {
+        container(name: 'kaniko') {
```
Comment (jglick) [inline src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/bourneShellElsewhereInPath.groovy]: Without the `ContainerExecDecorator` patch, this fails unless you add
```suggestion
        container(name: 'kaniko', shell: '/busybox/sh') {
```

File: src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/bourneShellElsewhereInPath.groovy [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+            sh 'ls -la /'
```
Comment (jglick) [inline src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/bourneShellElsewhereInPath.groovy]: Without the `BourneShellScript` patch, this fails unless you add
```suggestion
            sh '''#!/busybox/sh -ex
                ls -la /
            '''
```

**Comment by Vlatombe** [comment]
I'll hold this until there is a release of `durable-task-plugin`.

File: pom.xml [hunk 4]
```diff
@@ -94,10 +94,10 @@
       <version>6.8</version>
     </dependency>
 
-    <dependency> <!-- OnceRetentionStrategy -->
+    <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>durable-task</artifactId>
-      <version>1.29</version>
+      <version>1.30-rc416.ace2b55005b1</version> <!-- TODO https://github.com/jenkinsci/durable-task-plugin/pull/95 -->
```
Comment (dwnusbaum) [inline pom.xml]:
```suggestion
      <version>1.30</version>
```

**Pull request merged**